### ISSUE TITLE
ValidationException is raised for non-empty fields

### ISSUE THREAD

**Issue by tq0** [elastic/elasticsearch-dsl-py#418]
When using `required=True` with numeric, string or boolean type fields Field.clean method (https://github.com/elastic/elasticsearch-dsl-py/blob/master/elasticsearch_dsl/field.py#L81) raises `ValidationException`. There is a `FIXME` in the code for this already, but the issue is not related only to numeric types.

**Comment by HonzaKral** [comment]
Thanks for the report, this is still not fixed for numerical fields.

For string fields I am not sure what the correct behavior is, should we treat `''` as an acceptable value when the field is required? What do you think?

**Comment by tq0** [comment]
Thank you for your prompt reply!

In my opinion, `required=True` should check (e.g. when saving an instance of `DocType`) that a kwarg with that name has been provided. Also since [elasticsearch](https://www.elastic.co/guide/en/elasticsearch/guide/current/_dealing_with_null_values.html) treats `null` values as missing, I believe that `required=True` should also prevent that the provided kwarg equals `None`.

As far as acceptable values are considered, I think that there should be another parameter - something like `allow_blank` that would validate that `String` or `Nested` fields are not empty (have zero length).

To summarize, I would expect that the `required=True` instruction an raises exception if the parameter for that field has not been provided or its value is `None`. An option to check for empty values is in my opinion outside the scope of this issue and should perhaps be considered as a future improvement.
